src/runtime/stream_io.rs

```diff
@@ -30,6 +30,7 @@ pub struct StreamInput {
     name: String,
     item_size: usize,
     type_id: TypeId,
+    pub(crate) type_name: &'static str,
```

Comment:
Could create a `pub(crate)` accessor instead. But this should be crate-private to keep users from expecting this to be a stable identifier (which its not).